src/Navigation/Extra.elm

```diff
@@ -38,4 +38,8 @@ resolve : Navigation.Location -> String -> Navigation.Location
 resolve base url =
     locationFromString url
         -- TODO: implment correct logic (current logic is only correct for "authority-relative" URLs without query or fragment strings)
-        |> Maybe.withDefault { base | pathname = url }
+        |> Maybe.withDefault
+            { base
+                | href = (base.href |> String.split "/" |> List.take 3 |> String.join "/") ++ url
```

Comment:
is this to select only `http://blah.com/`? A comment might be useful there, but this is a nit.

Reply:
The relevant comment here, I think, is the one at the top of the module
> this module should implement the algorithm described at <https://url.spec.whatwg.org/>
:)